### PR TITLE
Custom return value delegate sequences for variadic functions

### PR DESCRIPTION
**Pull request by oliviera9**
This modification adds sequenced support to custom delegates for variadic functions.

### CODE REVIEW

**Comment by oliviera9** [comment]
I added the tests for the variadic functions delegate sequences feature.

File: test/test_cases.include [hunk 1]
```diff
@@ -353,6 +353,72 @@ TEST_F(FFFTestSuite, when_value_custom_fake_called_THEN_it_returns_custom_return
...
+  while((arg = va_arg(vl, int)) != 0)
+    a *= arg;
+  return a;
+}
+
+
+TEST_F(FFFTestSuite, vararg_custom_fake_sequence_not_exausthed)
+{
+  int (*custom_fakes[])(const char *, int, va_list) = {valuefunc3var_custom_fake1,
+                                    valuefunc3var_custom_fake2,
```
Comment (wulfgarpro) [inline test/test_cases.include]: indent

File: test/test_cases.include [hunk 2]
```diff
@@ -353,6 +353,72 @@ TEST_F(FFFTestSuite, when_value_custom_fake_called_THEN_it_returns_custom_return
...
+  int a = 1;
+  ASSERT_EQ(valuefunc3var("a", a, 2, 3, 4, 0), 10);
+  ASSERT_EQ(valuefunc3var("aa", a, 2, 3, 4, 2, 0), -10);
+  ASSERT_EQ(valuefunc3var("aaa", a, 2, 3, 0), 6);
+}
+
+TEST_F(FFFTestSuite, vararg_custom_fake_seq_return_values_saved_in_history)
+{
+  int (*custom_fakes[])(const char *, int, va_list) = {valuefunc3var_custom_fake1,
+                                    valuefunc3var_custom_fake2,
```
Comment (wulfgarpro) [inline test/test_cases.include]: indent

File: test/test_cases.include [hunk 3]
```diff
@@ -353,6 +353,72 @@ TEST_F(FFFTestSuite, when_value_custom_fake_called_THEN_it_returns_custom_return
...
+  valuefunc3var("aaa", a, 2, 3, 0);
+  ASSERT_EQ(10, valuefunc3var_fake.return_val_history[0]);
+  ASSERT_EQ(-10, valuefunc3var_fake.return_val_history[1]);
+  ASSERT_EQ(6, valuefunc3var_fake.return_val_history[2]);
+}
+
+TEST_F(FFFTestSuite, vararg_custom_fake_sequence_exhausted)
+{
+  int (*custom_fakes[])(const char *, int, va_list) = {valuefunc3var_custom_fake1,
+                                    valuefunc3var_custom_fake2,
```
Comment (wulfgarpro) [inline test/test_cases.include]: indent

**Comment by oliviera9** [comment]
I've done the adjustments you requested @wulfgarpro . Take a look and let me know :-)

**Pull request merged**